ethcore/src/engines/tendermint/mod.rs

```diff
@@ -103,6 +103,7 @@ pub struct Tendermint {
 
 struct EpochVerifier {
 	subchain_validators: SimpleList,
+	recover: Box<Fn(&Signature, &Message) -> Result<Address, Error> + Send + Sync>,
```

Comment:
Can we make this a generic parameter instead or even one thing in `cfg(test)` and another in `cfg(not(test))`? Doing several virtual calls per block will probably be a performance hit we don't have to take.